en/views/helpers/form.rst

```diff
@@ -164,43 +164,45 @@ Valid values:
 
 .. _form-values-from-query-string:
 
-Getting form values from the query string
+Getting form values from other values sources
 -----------------------------------------
 
 A FormHelper's values sources define where its rendered elements, such as
 input-tags, receive their values from.
 
-By default FormHelper draws its values from the 'context'.  The default
-contexts, such as ``EntityContext``, will fetch data from ``$request->getData()``
-or from the current entity.
+The supported sources are ``context``, ``data`` and ``query``. You can use one
+or more sources by setting ``valueSources`` option or by using ``setValuesSource()``.
+Any widgets generated by ``FormHelper`` will gather their values from the sources,
+in the order you setup.
+
+By default ``FormHelper`` draws its values from ``data`` or ``context``, i.e. it will
+fetch data from ``$request->getData()`` or, if not present, from the active context's
+data, that are the entity's data in the case of ``EntityContext``.
 
-If however, you are building a form that needs to read from the query string,
-you can use ``setValuesSource()`` to change where ``FormHelper`` reads data input
-data from::
+If however, you are building a form that needs to read from the query string, you can
+change where ``FormHelper`` reads input data from::
 
-    // Prioritize query string over context:
+    // Use query string instead of request data:
     echo $this->Form->create($article, [
+        'type' => 'get',
         'valueSources' => ['query', 'context']
     ]);
 
     // Same effect:
     echo $this->Form
-        ->setValueSources(['query', 'context'])
-        ->create($articles);
+        ->create($articles, ['type' => 'get'])
+        ->setValueSources(['query', 'context']);
```

Comment:
I like the PR in general. 
But are you sure that switching set and create works here?
Afaik set is charitable, but create() already returns the final string of HTML.

Reply:
Ops sorry, my mistake!

I'm used to the fact that create() methods usually return an instance for fluent interface :-) (it would make more sense to use ->print() or ->output() at the end).